gen/nvml/device.go

```diff
@@ -1468,7 +1468,7 @@ func (Device Device) GetGridLicensableFeatures() (GridLicensableFeatures, Return
 func DeviceGetProcessUtilization(Device Device, LastSeenTimeStamp uint64) ([]ProcessUtilizationSample, Return) {
 	var ProcessSamplesCount uint32
 	ret := nvmlDeviceGetProcessUtilization(Device, nil, &ProcessSamplesCount, LastSeenTimeStamp)
-	if ret != SUCCESS {
+	if ret != ERROR_INSUFFICIENT_SIZE {
```

Comment:
I had overloaded my suggested change in #11 without pointing this out explicitly:
```
	return Utilization[:ProcessSamplesCount], ret
```
But we should do that too.

Reply:
OK, so the second call to `nvmlDeviceGetProcessUtilization` may also modify the `ProcessSamplesCount`? Or at the very least reduce it?

Reply:
At least according to the API, yes.

Reply:
The docs:
```
 * On successful return, the function updates \a processSamplesCount with the number of process utilization sample
 * structures that were actually written. This may differ from a previously read value as instances are created or
 * destroyed.
 ```
don't seem to state that this is guaranteed to be less than the size of the input array, in which case I assume that returning `Utilization[:ProcessSamplesCount]` may be dangerous? Is it overkill to use `Utilization[:min(ProcessSamplesCount, len(Utilization)]` (or a more idiomatic version of this)?

Reply:
It should be implied because it is a C API. The whole point of the dance to first get the sample count and then re-call the function with an appropriately allocated array is so that you have the chance to allocate an array that is big enough to hold all the samples.